src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/defendant/ResponseDeadlineExtendedDashboardNotificationHandler.java

```diff
@@ -0,0 +1,64 @@
+package uk.gov.hmcts.reform.civil.handler.callback.camunda.dashboardnotifications.defendant;
+
+import static uk.gov.hmcts.reform.civil.callback.CallbackParams.Params.BEARER_TOKEN;
+import static uk.gov.hmcts.reform.civil.callback.CallbackType.ABOUT_TO_SUBMIT;
+import static uk.gov.hmcts.reform.civil.callback.CaseEvent.CREATE_CLAIMANT_DASHBOARD_NOTIFICATION_DEFENDANT_RESPONSE_DATE;
+import static uk.gov.hmcts.reform.civil.handler.callback.camunda.dashboardnotifications.DashboardScenarios.SCENARIO_AAA7_DEFENDANT_RESPONSE_MORE_TIME_REQUESTED_CLAIMANT;
+
+import java.util.List;
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackResponse;
+import uk.gov.hmcts.reform.civil.callback.Callback;
+import uk.gov.hmcts.reform.civil.callback.CallbackHandler;
+import uk.gov.hmcts.reform.civil.callback.CallbackParams;
+import uk.gov.hmcts.reform.civil.callback.CaseEvent;
+import uk.gov.hmcts.reform.civil.client.DashboardApiClient;
+import uk.gov.hmcts.reform.civil.model.CaseData;
+import uk.gov.hmcts.reform.civil.service.DashboardNotificationsParamsMapper;
+import uk.gov.hmcts.reform.dashboard.data.ScenarioRequestParams;
+
+@Service
+@RequiredArgsConstructor
+public class ResponseDeadlineExtendedDashboardNotificationHandler extends CallbackHandler {
+
+    private static final List<CaseEvent> EVENTS = List.of(CREATE_CLAIMANT_DASHBOARD_NOTIFICATION_DEFENDANT_RESPONSE_DATE);
+    public static final String TASK_ID = "GenerateClaimantDashboardNotificationClaimantNewResponseDeadline";
+    private final DashboardApiClient dashboardApiClient;
+    private final DashboardNotificationsParamsMapper mapper;
+
+    @Override
+    protected Map<String, Callback> callbacks() {
+        return Map.of(
+            callbackKey(ABOUT_TO_SUBMIT), this::configureScenarioForDefendantResponse
+        );
+    }
+
+    @Override
+    public String camundaActivityId(CallbackParams callbackParams) {
+        return TASK_ID;
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    private CallbackResponse configureScenarioForDefendantResponse(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        String authToken = callbackParams.getParams().get(BEARER_TOKEN).toString();
+        if (caseData.isRespondent1NotRepresented()) {
```

Comment:
As the only change depending on the condition is the scenario I think is better to a getScenario method that return a scenario depending on the conditions like we are doing here:
https://github.com/hmcts/civil-service/pull/4235/files

So we will only have one recordScenario

Reply:
Method added to get the correct scenario.  Currently it just returns the single one we have for this journey.